lib/mixin.js

```diff
@@ -0,0 +1,44 @@
+var Element = require('./Element')
+var mapValues = require('./utils/mapValues')
+
+var mixin = {
+  componentWillMount: function () {
+    this.connectedFauxDOM = {}
+    this.animateFauxDOMUntil = 0
+  },
+  connectFauxDOM: function (node, name) {
+    this.connectedFauxDOM[name] = typeof node !== 'string' ? node : new Element(node)
+    setTimeout(this.drawFauxDOM)
+    return this.connectedFauxDOM[name]
+  },
+  drawFauxDOM: function () {
+    var virtualDOM = mapValues(this.connectedFauxDOM, function (n) {
+      return n.toReact()
+    })
+    this.setState(virtualDOM)
+  },
+  animateFauxDOM: function (duration) {
+    this.animateFauxDOMUntil = Math.max(Date.now() + duration, this.animateFauxDOMUntil)
```

Comment:
You're probably right. `Connect`ing more than one element is probably an edge case as is, anyway. And the performance win would be very small since React makes the comparison awfully quick.